### PR TITLE
Fix line retainAll doesn't modify names and values array

### PR DESCRIPTION
**Pull request by Linary**
Change-Id: I383b3b20a162816098aa8ee8468e8255b3553fdc

### CODE REVIEW

**Comment by Linary** [comment]
fix #108

File: src/main/java/com/baidu/hugegraph/loader/reader/Line.java [hunk 1]
```diff
@@ -29,65 +29,59 @@
     private final String rawLine;
     private String[] names;
     private Object[] values;
-    private Map<String, Object> keyValues;
+    private final Map<String, Object> keyValues;
 
     public Line(String rawLine, Map<String, Object> keyValues) {
         this.rawLine = rawLine;
```
Comment (zhoney) [inline src/main/java/com/baidu/hugegraph/loader/reader/Line.java]: check keyValues not null

File: src/main/java/com/baidu/hugegraph/loader/reader/Line.java [hunk 2]
```diff
@@ -22,72 +22,73 @@
...
+        this.values = getValues(keyValues, this.names);
     }
 
     public Line(String rawLine, String[] names, Object[] values) {
-        assert names.length == values.length;
+        E.checkArgumentNotNull(rawLine, "The rawLine can't be null");
+        E.checkArgumentNotNull(names, "The names can't be null");
+        E.checkArgumentNotNull(values, "The values can't be null");
+        E.checkArgument(names.length == values.length,
+                        "The length of names %s should same as values %s");
```
Comment (javeme) [inline src/main/java/com/baidu/hugegraph/loader/reader/Line.java]: should be same

File: src/main/java/com/baidu/hugegraph/loader/reader/Line.java [hunk 3]
```diff
@@ -22,72 +22,73 @@
...
         this.rawLine = rawLine;
         this.names = names;
         this.values = values;
-        this.keyValues = null;
+        this.keyValues = InsertionOrderUtil.newMap();
+        for (int i = 0, n = this.names.length; i < n; i++) {
```
Comment (javeme) [inline src/main/java/com/baidu/hugegraph/loader/reader/Line.java]: just keep i < this.names.length

**Pull request merged**